server/src/main/scala/services/graphstages/ArrivalsGraphStage.scala

```diff
@@ -260,23 +260,97 @@ class ArrivalsGraphStage(name: String = "",
       mergeBestFieldsFromSources(baseArrival, mergeArrival(key).getOrElse(baseArrival))
     }
 
+
+    def arrivalMapForSourceType(searchSourceType: ArrivalsSourceType): SortedMap[UniqueArrival, Arrival] = searchSourceType match {
+      case LiveArrivals => liveArrivals
+      case LiveBaseArrivals => liveBaseArrivals
+      case BaseArrivals => forecastBaseArrivals
+    }
+
+    def potentialKeys(mergeKey: UniqueArrival, searchArrivalMap: SortedMap[UniqueArrival, Arrival]): Iterable[UniqueArrival] =
+      searchArrivalMap.keys.filter(_.potentialKey(mergeKey, 1 * 60 * 60 * 1000))
+
+    def mergeScheduledApproximation(mergeKey: UniqueArrival, mergeKeySourceType: ArrivalsSourceType, foundArrival: Arrival, searchSourceType: ArrivalsSourceType) = {
```

Comment:
Is `mergeKey` the UniqueArrival for `foundArrival`? If so we don't need to pass it as it's available as `unique` from the arrival itself